configuration/config2/SchemaTypeStringLimits.cpp

```diff
@@ -35,8 +35,8 @@ bool SchemaTypeStringLimits::doValueTest(const std::string &testValue) const
     // test patterns
     for (auto it = m_patterns.begin(); it != m_patterns.end() && isValid; ++it)
     {
-        std::regex expr ((*it).c_str());
-        isValid = std::regex_match(testValue, expr);
+        static const boost::regex e(*it);
+        isValid = boost::regex_match(testValue, e);
```

Comment:
isValid is overwritten on every iteration

Reply:
The loop has a test for isValid being false and will exit.